pkg/controller/vitessbackupschedule/vitessbackupschedule_controller.go

```diff
@@ -0,0 +1,579 @@
+		if jobs[i].Status.StartTime == nil {
+			return jobs[j].Status.StartTime != nil
+		}
+		return jobs[i].Status.StartTime.Before(jobs[j].Status.StartTime)
+	})
+
+	for i, job := range jobs {
+		if int32(i) >= int32(len(jobs))-limit {
+			break
+		}
+		if err := r.client.Delete(ctx, job, client.PropagationPolicy(metav1.DeletePropagationBackground)); (err) != nil {
+			log.WithError(err).Errorf("unable to delete old job: %s", job.Name)
+		} else {
+			log.Infof("deleted old job: %s", job.Name)
+		}
+	}
+}
+
+func (r *ReconcileVitessBackupsSchedule) removeTimeoutJobs(ctx context.Context, jobs []*kbatch.Job, timeout int32) error {
+	if timeout == -1 {
+		return nil
+	}
+	for _, job := range jobs {
+		jobStartTime, err := getScheduledTimeForJob(job)
+		if err != nil {
+			return err
+		}
+		if jobStartTime.After(time.Now().Add(time.Minute * time.Duration(timeout))) {
+			if err := r.client.Delete(ctx, job, client.PropagationPolicy(metav1.DeletePropagationBackground)); (err) != nil {
+				log.WithError(err).Errorf("unable to delete timed out job: %s", job.Name)
+			} else {
+				log.Infof("deleted timed out job: %s", job.Name)
+			}
+		}
+	}
+	return nil
+}
+
+func isJobFinished(job *kbatch.Job) (bool, kbatch.JobConditionType) {
+	for _, c := range job.Status.Conditions {
+		if (c.Type == kbatch.JobComplete || c.Type == kbatch.JobFailed) && c.Status == corev1.ConditionTrue {
+			return true, c.Type
+		}
+	}
+
+	return false, ""
+}
+
+func getScheduledTimeForJob(job *kbatch.Job) (*time.Time, error) {
+	timeRaw := job.Annotations[scheduledTimeAnnotation]
+	if len(timeRaw) == 0 {
+		return nil, nil
+	}
+
+	timeParsed, err := time.Parse(time.RFC3339, timeRaw)
+	if err != nil {
+		return nil, err
+	}
+
+	return &timeParsed, nil
+}
+
+func (r *ReconcileVitessBackupsSchedule) createJob(ctx context.Context, vbsc *planetscalev2.VitessBackupSchedule, scheduledTime time.Time) (*kbatch.Job, error) {
+	name := fmt.Sprintf("%s-%d", vbsc.Name, scheduledTime.Unix())
+
+	meta := metav1.ObjectMeta{
+		Labels: map[string]string{
+			planetscalev2.BackupScheduleLabel: vbsc.Name,
+		},
+		Annotations: make(map[string]string),
+		Name:        name,
+		Namespace:   vbsc.Namespace,
+	}
+	for k, v := range vbsc.Annotations {
+		meta.Annotations[k] = v
+	}
+	meta.Annotations[scheduledTimeAnnotation] = scheduledTime.Format(time.RFC3339)
+
+	for k, v := range vbsc.Labels {
+		meta.Labels[k] = v
+	}
+
+	pod, err := r.createJobPod(ctx, vbsc, name)
+	if err != nil {
+		return nil, err
+	}
+	job := &kbatch.Job{
+		ObjectMeta: meta,
+		Spec: kbatch.JobSpec{
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: meta,
+				Spec:       pod,
+			},
+		},
+	}
+
+	if err := ctrl.SetControllerReference(vbsc, job, r.scheme); err != nil {
+		return nil, err
+	}
+
+	return job, nil
+}
+
+func (r *ReconcileVitessBackupsSchedule) createJobPod(ctx context.Context, vbsc *planetscalev2.VitessBackupSchedule, name string) (pod corev1.PodSpec, err error) {
```

Comment:
have we taking the vtctld action timeout into account?

Reply:
I do not think so, what are you referring to?

Reply:
or make it a flag, like `extraFlags`, on the backup schedule

Reply:
I think this is solved as people can set `--action-timeout` by adding it to the `extraFlags` field of `VitessBackupScheduleStrategy`.

Reply:
might be worth adding a note about that in release notes. i expect it will be a common issue people run in to.

Reply:
fixed via [9739a94](https://github.com/planetscale/vitess-operator/pull/553/commits/9739a94314027de8edf1451a07528f2414d4bca4)